src/gs-shell.c

```diff
@@ -924,6 +924,16 @@ side_filter_row_selected (GtkListBox *side_filter,
 	}
 }
 
+static gboolean
+side_filter_row_has_category (GsSideFilterRow *row,
+			      GsCategory *category)
+{
+	if (!category)
```

Comment:
Is this necessary? Other code assumes that when `mode == GS_SHELL_MODE_CATEGORY`, `gs_shell_category_get_category (priv->shell_category)` will return a category